Assets/Scripts/EnemyProjectile.cs

```diff
@@ -41,17 +42,23 @@ void OnCollisionEnter(Collision collision)
 		Destroy(gameObject);
     }
 
-	public void Launch(Vector3 playerVelocity)
+	public void Launch(Vector3 playerVelocity, float maxDistance)
 	{
-		GetComponent<Rigidbody>().velocity = CalculateLaunchData(playerVelocity).initialVelocity;
+		GetComponent<Rigidbody>().velocity = CalculateLaunchData(playerVelocity, maxDistance).initialVelocity;
 	}
 
-	LaunchData CalculateLaunchData(Vector3 playerVelocity)
+	private static float map(float value, float fromLow, float fromHigh, float toLow, float toHigh)
+	{
+		return (value - fromLow) * (toHigh - toLow) / (fromHigh - fromLow) + toLow;
+	}
```

Comment:
Why do we need this? Could we specify the name of the method to communicate its purpose?

Reply:
Its a standard mapping function, C# just doesnt have one by default apperently

Reply:
what does it map to though? Just want some explanation 😅